clients/venice-client/src/main/java/com/linkedin/venice/fastclient/RetriableAvroGenericStoreClient.java

```diff
@@ -198,7 +199,8 @@ public void onCompletion(Optional<Exception> exception) {
         if (exception.isPresent()) {
           streamingResponseFuture.completeExceptionally(exception.get());
         } else {
-          streamingResponseFuture.complete(new VeniceResponseMapImpl<>(valueMap, nonExistingKeys, true));
+          streamingResponseFuture
+              .complete(new VeniceResponseMapImpl<>(valueMap, nonExistingKeys, valueMap.size() == keys.size()));
```

Comment:
Even though batchGet uses `streamingBatchGet` (needs to be enabled for now), `batchGet` API still returns `CompletableFuture<Map<K, V>>` and if one of the routes takes a long time, it still slows down the entire batchGet before being used and currently returns an exception if the returned results is partial. Is my understanding of the API correct here? @xunyin8

Reply:
You meant server instead of router right? Good find, I think this is just an incomplete implementation of stream batch get in FC. We want to support partial batch get result just like what we do in thin-client.

Reply:
as per the discussion yesterday, batchGet() only returns full response or exception. So its inline.